src/node_api.cc

```diff
@@ -0,0 +1,2430 @@
+      v8::Number::New(v8impl::V8IsolateFromJsEnv(e), v));
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_create_boolean(napi_env e, bool b, napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  *result = v8impl::JsValueFromV8LocalValue(
+      v8::Boolean::New(v8impl::V8IsolateFromJsEnv(e), b));
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_create_symbol(napi_env e, const char* s, napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  v8::Isolate* isolate = v8impl::V8IsolateFromJsEnv(e);
+  if (s == nullptr) {
+    *result = v8impl::JsValueFromV8LocalValue(v8::Symbol::New(isolate));
+  } else {
+    v8::Local<v8::String> string;
+    CHECK_NEW_FROM_UTF8(isolate, string, s);
+
+    *result = v8impl::JsValueFromV8LocalValue(v8::Symbol::New(isolate, string));
+  }
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_create_error(napi_env e, napi_value msg, napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  *result = v8impl::JsValueFromV8LocalValue(v8::Exception::Error(
+      v8impl::V8LocalValueFromJsValue(msg).As<v8::String>()));
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_create_type_error(napi_env e,
+                                   napi_value msg,
+                                   napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  *result = v8impl::JsValueFromV8LocalValue(v8::Exception::TypeError(
+      v8impl::V8LocalValueFromJsValue(msg).As<v8::String>()));
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_create_range_error(napi_env e,
+                                    napi_value msg,
+                                    napi_value* result) {
+  NAPI_PREAMBLE(e);
+  CHECK_ARG(result);
+
+  *result = v8impl::JsValueFromV8LocalValue(v8::Exception::RangeError(
+      v8impl::V8LocalValueFromJsValue(msg).As<v8::String>()));
+
+  return GET_RETURN_STATUS();
+}
+
+napi_status napi_get_type_of_value(napi_env e,
+                                   napi_value value,
+                                   napi_valuetype* result) {
+  // Omit NAPI_PREAMBLE and GET_RETURN_STATUS because V8 calls here cannot throw
+  // JS exceptions.
+  CHECK_ARG(result);
+
+  v8::Local<v8::Value> v = v8impl::V8LocalValueFromJsValue(value);
+
+  if (v->IsNumber()) {
+    *result = napi_number;
+  } else if (v->IsString()) {
+    *result = napi_string;
+  } else if (v->IsFunction()) {
+    // This test has to come before IsObject because IsFunction
+    // implies IsObject
+    *result = napi_function;
+  } else if (v->IsObject()) {
+    *result = napi_object;
+  } else if (v->IsBoolean()) {
+    *result = napi_boolean;
+  } else if (v->IsUndefined()) {
+    *result = napi_undefined;
+  } else if (v->IsSymbol()) {
+    *result = napi_symbol;
+  } else if (v->IsNull()) {
+    *result = napi_null;
+  } else if (v->IsExternal()) {
+    *result = napi_external;
+  } else {
+    *result = napi_object;  // Is this correct?
```

Comment:
It should return an error in this case - Resolved by https://github.com/nodejs/node/pull/11975/commits/6b4ce5389dbb9642ae5f22ecc2cb2c066913d451